# HG changeset patch
# User Jean-Francois Dockes <jfd@recoll.org>
# Date 1381232776 -7200
# Node ID 65a7041abbe6e64f6597d5cf27690e91014a85d4
# Parent  0adf8145feafe2808202ba8c4cd73e4bac66acd8
autoconfig out things that dont work currently on mac os x

diff --git a/src/configure.ac b/src/configure.ac
--- configure.ac
+++ configure.ac
@@ -21,6 +21,12 @@
 fi
 (cd mk; rm -f sysconf; ln -s $sys sysconf)
 
+# There a few Recoll users on Mac OS X and a few things are just not worth
+# supporting
+if test X$sys = XDarwin ; then
+   NODYNLIB=#
+fi
+
 AC_SYS_LARGEFILE
 
 # OpenBSD needs sys/param.h for mount.h to compile
@@ -219,6 +225,7 @@
 #   configure --disable-python-module; make; make install;make clean
 #   configure; make; cd python/recoll; make install
 #
+if test X$sys != XDarwin ; then
 AC_ARG_ENABLE(python-module,
     AC_HELP_STRING([--disable-python-module],
    [Do not build the Python module.]),
@@ -228,6 +235,9 @@
 else
   NOPYTHON="#"
 fi
+else
+  NOPYTHON="#"
+fi
 
 # Build PIC objects for the library ?
 AC_ARG_ENABLE(pic,
@@ -562,6 +572,7 @@
 AC_SUBST(NOPIC)
 AC_SUBST(NOTHREADS)
 AC_SUBST(NOPYTHON)
+AC_SUBST(NODYNLIB)
 AC_SUBST(RCLVERSION)
 AC_SUBST(RCLLIBVERSION)
 
diff --git a/src/recollinstall.in b/src/recollinstall.in
--- recollinstall.in
+++ recollinstall.in
@@ -117,7 +117,8 @@
 ${INSTALL} -m 0755 index/recollindex ${bindir} || exit 1
 ${STRIP} ${bindir}/recollindex
 
-${INSTALL} -m 0644 lib/librecoll.so.${RCLLIBVERSION} ${libdir}/recoll/ || exit 1
+@NODYNLIB@${INSTALL} -m 0644 lib/librecoll.so.${RCLLIBVERSION} \
+@NODYNLIB@    ${libdir}/recoll/ || exit 1
 
 ${INSTALL} -m 0755 filters/rcl* ${datadir}/recoll/filters/ || exit 1
 for f in rclexecm.py rcllatinstops.zip;do
